modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js

```diff
@@ -43,65 +47,87 @@ class DateTimeInput extends React.Component {
-
+	_handleDateChange = (value) => {
 		this.setState({value});
 	};
 
-	_handleDateBlur = (event) => {
-		const date = dateFns.format(event.target.value, INPUT_DATE_FORMAT);
-
-		if (date !== 'Invalid Date') {
-			this.setState(
-				{
-					value: date,
-				},
-				() => {
-					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(date, INPUT_DATE_FORMAT)
-							.toISOString(),
-					});
-				}
-			);
+	_handleExpandedChange = (expandedState) => {
+		this.setState({expanded: expandedState});
+
+		if (expandedState === false) {
+			this._saveDateTimeValue();
 		}
-		else {
-			const resetDate = dateFns.format(new Date(), INPUT_DATE_FORMAT);
-
-			this.setState(
-				{
-					value: resetDate,
-				},
-				() => {
-					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(resetDate, INPUT_DATE_FORMAT)
-							.toISOString(),
-					});
-				}
-			);
+	};
+
+	_saveDateTimeValue = () => {
+		const date = dateFns.format(this.state.value, INPUT_DATE_FORMAT);
+
+		if (this.previousValue !== date) {
+			if (date !== 'Invalid Date') {
+				this.setState(
+					{
+						value: date,
+					},
+					() => {
+						this.props.onChange({
+							type: PROPERTY_TYPES.DATE_TIME,
+							value: dateFns
+								.parse(date, INPUT_DATE_FORMAT)
+								.toISOString(),
+						});
+					}
+				);
+			}
+			else {
+				const resetDate = dateFns.format(new Date(), INPUT_DATE_FORMAT);
+
+				this.setState(
+					{
+						value: resetDate,
+					},
+					() => {
+						this.props.onChange({
+							type: PROPERTY_TYPES.DATE_TIME,
+							value: dateFns
+								.parse(resetDate, INPUT_DATE_FORMAT)
+								.toISOString(),
+						});
+					}
+				);
+			}
 		}
+		this.previousValue = date;
 	};
 
 	render() {
-		const {value} = this.state;
+		const {expanded, value} = this.state;
 		const {disabled, propertyLabel} = this.props;
 
 		return (
 			<div className="criterion-input date-input">
-				<input
+				<ClayDatePicker
 					aria-label={`${propertyLabel}: ${Liferay.Language.get(
 						'select-date'
 					)}`}
-					className="form-control"
-					data-testid="date-input"
 					disabled={disabled}
-					onBlur={this._handleDateBlur}
+					expanded={expanded}
+					months={[
+						`${Liferay.Language.get('january')}`,
+						`${Liferay.Language.get('february')}`,
+						`${Liferay.Language.get('march')}`,
+						`${Liferay.Language.get('april')}`,
+						`${Liferay.Language.get('may')}`,
+						`${Liferay.Language.get('june')}`,
+						`${Liferay.Language.get('july')}`,
+						`${Liferay.Language.get('august')}`,
+						`${Liferay.Language.get('september')}`,
+						`${Liferay.Language.get('october')}`,
+						`${Liferay.Language.get('november')}`,
+						`${Liferay.Language.get('december')}`,
+					]}
+					onBlur={this._saveDateTimeValue}
 					onChange={this._handleDateChange}
-					type="date"
+					onExpandedChange={this._handleExpandedChange}
```

Comment:
Do we need the `onExpendedChange` if we are using the `onBlur` ? I don't know if we can get rid of the `onExpendedChange`

Reply:
Yes we need it because when we open the calendar the focus doesn't go to the input. The onBlur is for the cases the user sets a value using the keyboard and the OnExpendedChange is for when the user sets the value by opening the calendar and selecting